app/models/document.rb

```diff
@@ -39,6 +40,9 @@ class Document < ApplicationRecord
   has_many :document_collections, through: :document_collection_groups
   has_many :features, inverse_of: :document, dependent: :destroy
 
+  #binding.pry
+  before_update :check_if_locked_document_internal
```

Comment:
I'm not too keen on the name here (perhaps consider removing the `_internal` part and giving it a unique name)?